tests/test_subscriptions.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import json
+import logging
+import os
+import sys
+import unittest
+import uuid
+from io import open
+
+import google.auth
+import google.auth.transport.requests
+import requests
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..')) # noqa
+sys.path.insert(0, pkg_root) # noqa
+
+import dss
+from dss import DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE, DSS_ELASTICSEARCH_QUERY_TYPE
+from dss.config import DeploymentStage, override_email_config
+from dss.util import UrlBuilder
+from dss.util.es import ElasticsearchClient, get_elasticsearch_index
+from tests.es import check_start_elasticsearch_service, close_elasticsearch_connections, elasticsearch_delete_index
+from tests.infra import DSSAsserts
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class TestSubscriptions(unittest.TestCase, DSSAsserts):
+
+    def setUp(self):
+        os.environ['DSS_ES_ENDPOINT'] = os.getenv('DSS_ES_ENDPOINT', "localhost")
+
+        self.app = dss.create_app().app.test_client()
+        dss.Config.set_config(dss.DeploymentStage.TEST)
+
+        logger.debug("Setting up Elasticsearch")
+        es_client = ElasticsearchClient.get(logger)
+        es_client.indices.delete(index="_all", ignore=[404])  # Disregard if no indices - don't error.
+        index_mapping = {  # Need to make a mapping for the percolator type before we add any.
+            "mappings": {
+                DSS_ELASTICSEARCH_QUERY_TYPE: {
+                    "properties": {
+                        "query": {
+                            "type": "percolator"
+                        }
+                    }
+                }
+            }
+        }
+        get_elasticsearch_index(es_client, DSS_ELASTICSEARCH_INDEX_NAME, logger, index_mapping)
+
+        with open(os.path.join(os.path.dirname(__file__), "sample_index_doc.json"), "r") as fh:
+            index_document = json.load(fh)
+
+        self.callback_url = "https://example.com"
+        with open(os.path.join(os.path.dirname(__file__), "sample_query.json"), "r") as fh:
+            self.sample_percolate_query = json.load(fh)
+
+        es_client.index(index=DSS_ELASTICSEARCH_INDEX_NAME,
+                        doc_type=DSS_ELASTICSEARCH_DOC_TYPE,
+                        id=str(uuid.uuid4()),
+                        body=index_document,
+                        refresh=True)
+
+    def test_put(self):
+        uuid_ = self._put_subscription()
+
+        es_client = ElasticsearchClient.get(logger)
+        response = es_client.get(index=DSS_ELASTICSEARCH_INDEX_NAME,
+                                 doc_type=DSS_ELASTICSEARCH_QUERY_TYPE,
+                                 id=uuid_)
+        registered_query = response['_source']
+        self.assertEqual(self.sample_percolate_query, registered_query)
+
+    def test_get(self):
```

Comment:
should test your negative code paths like:

1. can't find it.
2. not the owner.